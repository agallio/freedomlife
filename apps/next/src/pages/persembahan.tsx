import { NextPage } from 'next'
import Head from 'next/head'
import Image from 'next/image'
import { NextSeo } from 'next-seo'
import toast from 'react-hot-toast'

// Components
import JumboHeader from '~/components/JumboHeader'

// Icon Components
import CopyIcon from '~/components/Icons/CopyIcon'

// Utils
import { checkTheme } from '~/utils/hooks/useDynamicTheme'

const Persembahan: NextPage = () => {
  const theme = checkTheme()

  const fallbackCopyNumber = async (bank: string, bankText: string) => {
    const textArea = document.createElement('textarea')
    textArea.value = bank

    textArea.style.top = '0'
    textArea.style.left = '0'
    textArea.style.position = 'fixed'

    document.body.appendChild(textArea)
    textArea.focus()
    textArea.select()

    try {
      await document.execCommand('copy')
      textArea.style.display = 'none'
      toast.success(`Nomor rekening ${bankText} tersalin!`, {
        style:
          theme === 'dark' ? { background: '#111827', color: '#ffffff' } : {},
      })
    } catch (e) {
      console.error(e)
      toast.error('Gagal menyalin nomor, coba sesaat lagi.', {
        style:
          theme === 'dark' ? { background: '#111827', color: '#ffffff' } : {},
      })
    }
  }

  const copyNumber = async (bank: string) => {
    const bankText = bank === 'cimb' ? 'CIMB Niaga' : 'BCA'
    if (!navigator.clipboard) {
      fallbackCopyNumber(
        bank === 'cimb' ? '800077521000' : '0171217007',
        bankText,
      )
      return
    }

    try {
      await navigator.clipboard.writeText(
        bank === 'cimb' ? '800077521000' : '0171217007',
      )
      toast.success(`Nomor rekening ${bankText} tersalin!`, {
        style:
          theme === 'dark' ? { background: '#111827', color: '#ffffff' } : {},
      })
    } catch (e) {
      console.error(e)
      toast.error('Gagal menyalin nomor, coba sesaat lagi.', {
        style:
          theme === 'dark' ? { background: '#111827', color: '#ffffff' } : {},
      })
    }
  }

  return (
    <>
      <Head>
        <title>Persembahan | freedomlife</title>
      </Head>
      <NextSeo
        noindex
        title="Persembahan | freedomlife"
        description="Halaman informasi persembahan dan perpuluhan Gereja Kristen Kemah Daud Yogyakarta."
        openGraph={{
          url: 'https://freedomlife.id/persembahan',
          title: 'Persembahan | freedomlife',
          description:
            'Halaman informasi persembahan dan perpuluhan Gereja Kristen Kemah Daud Yogyakarta.',
          site_name: 'freedomlife',
          images: [
            {
              url: 'https://freedomlife.id/images/og-persembahan.png',
              alt: `Tulisan dan logo 'freedomlife' disertai dengan keterangan: 'Persembahan dan Perpuluhan'`,
            },
          ],
        }}
        twitter={{
          cardType: 'summary_large_image',
        }}
      />

      <div className="mx-auto mb-6 max-w-sm p-6 sm:max-w-md sm:py-6 md:mb-16 landscape:mx-auto">
        <JumboHeader
          subtitle="Persembahan &amp; Persepuluhan"
          description="Gereja Kristen Kemah Daud - Yogyakarta"
        />

        <main className="mt-4">
          <div
            className="mt-4 flex flex-col rounded-lg shadow-md"
            style={{
              background:
                'linear-gradient(45deg, rgba(16,185,129,1) 30%, rgba(0,212,255,1) 100%)',
            }}
          >
            <p className="p-4 text-white">
              Dukungan untuk pekerjaan pelayanan Gereja Kristen Kemah Daud
              Yogyakarta dapat dilakukan melalui scan QRIS dibawah ini:
            </p>

            <Image
              src="/images/qris.jpeg"
              alt="QRIS Kemah Daud Yogyakarta"
              width={405}
              height={458}
            />

            <p className="p-4 text-white">
              Melalui aplikasi m-Banking / Digital Payment seperti:
            </p>

            <div className="flex px-4 pb-4 sm:px-12">
              <Image
                src="/images/ewallet.png"
                alt="All E-Wallet"
                width={1080}
                height={337}
              />
            </div>
          </div>

          <div className="mt-8 flex flex-col rounded-lg p-4 shadow-md dark:bg-gray-600">
            <p className="dark:text-white">
              Atau dapat juga dilakukan melalui transfer pada rekening bank
              dibawah ini:
            </p>

            <div className="mt-6">
              <p className="text-lg font-bold dark:text-white">CIMB Niaga</p>
              <p className="mb-2 text-gray-600 dark:text-gray-200">
                a/n Gereja Kristen Kemah Daud
              </p>
              <div className="flex w-full items-center justify-between rounded-lg border border-gray-300 px-4 py-2 dark:border-gray-400">
                <p className="dark:text-white">8000-7752-1000</p>
                <button
                  className="text-emerald-600 hover:text-emerald-800 focus:outline-none dark:text-emerald-400 dark:hover:text-emerald-500"
                  onClick={() => copyNumber('cimb')}
                >
                  <CopyIcon className="h-6 w-6" />
                </button>
              </div>
            </div>

            <div className="mb-6 mt-4">
              <p className="text-lg font-bold dark:text-white">BCA</p>
              <p className="mb-2 text-gray-600 dark:text-gray-200">
                a/n GKKD Glory of God Yogyakarta
              </p>
              <div className="flex w-full items-center justify-between rounded-lg border border-gray-300 px-4 py-2 dark:border-gray-400">
                <p className="dark:text-white">017-121-7007</p>
                <button
                  className="text-emerald-600 hover:text-emerald-800 focus:outline-none dark:text-emerald-400 dark:hover:text-emerald-500"
                  onClick={() => copyNumber('bca')}
                >
                  <CopyIcon className="h-6 w-6" />
                </button>
              </div>
            </div>
          </div>
        </main>
      </div>
    </>
  )
}

export default Persembahan
