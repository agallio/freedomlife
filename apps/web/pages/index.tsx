// Screen
import HomeScreen from '@repo/app/features/home'

// Icon Components
import FreedomLifeIcon from '@repo/app/components/icons/freedomlife-icon'

export default function HomePage() {
  return (
    <div className="mx-auto flex w-full max-w-sm flex-col px-6 pb-28 pt-4 sm:max-w-md">
      <div className="flex pb-4">
        <FreedomLifeIcon className="w-[230px]" />
      </div>

      <HomeScreen />
    </div>
  )
}
