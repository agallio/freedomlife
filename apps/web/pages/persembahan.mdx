import Head from 'next/head'
import { NextSeo } from 'next-seo'
import MdxLayout from '@repo/app/components/mdx-layout.web'

<Head>
  <title>Persembahan | freedomlife</title>
</Head>
<NextSeo
  noindex
  nofollow
  title="Persembahan | freedomlife"
  description="Halaman informasi persembahan dan perpuluhan Gereja Kristen Kemah Daud Yogyakarta."
  openGraph={{
    url: 'https://freedomlife.id/persembahan',
    title: 'Persembahan | freedomlife',
    description:
      'Halaman informasi persembahan dan perpuluhan Gereja Kristen Kemah Daud Yogyakarta.',
    site_name: 'freedomlife',
    images: [
      {
        url: 'https://freedomlife.id/images/og-persembahan.png',
        alt: `Tulisan dan logo 'freedomlife' disertai dengan keterangan: 'Persembahan dan Perpuluhan'`,
      },
    ],
  }}
  twitter={{
    cardType: 'summary_large_image',
  }}
/>

Dukungan untuk pekerjaan & pelayanan Gereja Kristen Kemah Daud Yogyakarta dapat dilakukan melalui QRIS dibawah ini.

<div className="flex items-center justify-center rounded-lg bg-white shadow-sm">
  <img src="/images/qris.jpeg" alt="QRIS GKKD Yogyakarta" width="300" />
</div>

Dapat digunakan di semua aplikasi yang mendukung QRIS di Indonesia.

<div className="flex flex-row items-center justify-center rounded-lg bg-white px-4 shadow-sm">
  <img src="/images/ewallet.png" alt="All e-wallet images" />
</div>

Dapat juga dilakukan melalu transfer pada rekening bank dibawah ini.

- **8000-7752-1000**

  CIMB Niaga (a/n Gereja Kristen Kemah Daud)

- **017-121-7007**

  BCA (a/n GKKD Glory of God Yogyakarta)

export default function MDXPage({ children }) {
  return (
    <MdxLayout
      header={
        <div className="pb-4 text-emerald-900 dark:text-white">
          <h1 className="text-2xl font-bold">Persembahan & Persepuluhan</h1>
          <p>Gereja Kristen Kemah Daud - Yogyakarta</p>
        </div>
      }
    >
      {children}
    </MdxLayout>
  )
}
