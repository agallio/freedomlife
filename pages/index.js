import React, { Component } from 'react';
import { connect } from 'react-redux';
import Router from 'next/router';
import {
  LinearProgress,
  Card,
  CardContent,
  Typography,
  Grid,
  Fab
} from '@material-ui/core';
import moment from 'moment';
import 'moment/locale/id';

import { fetchGuideToday } from '../src/actions/guide';
import { fetchTodayChapter } from '../src/actions/bible';

class Index extends Component {
  componentDidMount = async () => {
    if (this.props.guide.guideToday.date !== moment().format('DD-MM-YYYY')) {
      this.props.fetchGuideToday();
    }
  };

  toBible = () => {
    if (
      this.props.guide.guideToday.date === moment().format('DD-MM-YYYY') &&
      this.props.bible.chapters.passage.length === 0
    ) {
      this.props.fetchTodayChapter();
    }
    Router.push('/bible');
  };

  render() {
    const { isFetching, guideToday } = this.props.guide;

    return (
      <div>
        {isFetching && <LinearProgress color="secondary" />}
        <div className="container" style={{ paddingBottom: 90 }}>
          <h2 className="header-title" style={{ marginTop: 70 }}>
            Freedom Life
          </h2>
          <p className="header-subtitle">
            Aplikasi panduan baca Alkitab setahun
          </p>

          <div style={{ marginTop: 40 }}>
            <Card className="styled-card">
              <CardContent>
                <Typography className="bold-text primary" variant="h5">
                  Panduan Hari Ini
                </Typography>
                <Typography className="light-text primary" variant="subtitle1">
                  {moment().format('dddd, LL')}
                </Typography>

                {['PL', 'PB1', 'PB2'].map(item => (
                  <Grid
                    key={item}
                    container
                    direction="row"
                    alignItems="center"
                    spacing={2}
                    style={{ marginTop: 5 }}
                  >
                    <Grid item xs={3} sm={2} md={2}>
                      <div className="guide-passage-box">
                        <h5 className="guide-passage-box-text">{item}</h5>
                      </div>
                    </Grid>
                    <Grid item xs={9} sm={10} md={10}>
                      {isFetching ? (
                        <LinearProgress />
                      ) : (
                        <Typography
                          className="bold-text primary"
                          variant="h6"
                          style={{ fontSize: 17 }}
                        >
                          {item === 'PL'
                            ? guideToday.pl_name
                            : item === 'PB1'
                            ? guideToday.pb1_name
                            : item === 'PB2'
                            ? guideToday.pb2_name
                            : 'Tidak ada data'}
                        </Typography>
                      )}
                      <Typography
                        className="light-text primary"
                        variant="subtitle1"
                      >
                        {item === 'PL'
                          ? 'Perjanjian Lama'
                          : item === 'PB1'
                          ? 'Perjanjian Baru 1'
                          : item === 'PB2'
                          ? 'Perjanjian Baru 2'
                          : ''}
                      </Typography>
                    </Grid>
                  </Grid>
                ))}

                <br />
                <Fab
                  className="guide-passage-box-fab"
                  size="small"
                  variant="extended"
                  color="primary"
                  onClick={this.toBible}
                >
                  Baca
                </Fab>
              </CardContent>
            </Card>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    guide: state.guide,
    bible: state.bible
  };
};

const mapDispatchToProps = dispatch => {
  return {
    fetchGuideToday: () => dispatch(fetchGuideToday()),
    fetchTodayChapter: version => dispatch(fetchTodayChapter(version))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Index);
