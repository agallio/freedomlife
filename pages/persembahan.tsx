import { SyntheticEvent, useEffect, useState } from 'react'
import Head from 'next/head'
import { Card, CardContent, Button, Snackbar } from '@material-ui/core'
import { Alert } from '@material-ui/lab'
import { CheckCircle as SuccessIcon } from '@material-ui/icons'

export const Persembahan: React.FC = () => {
  const [copiedNumber, setCopiedNumber] = useState(false)
  const [error, setError] = useState(false)
  const [bank, setBank] = useState('')

  useEffect(() => {
    const timeoutId = setTimeout(() => {
      setCopiedNumber(false)
      setBank('')
    }, 2000)
    return () => clearTimeout(timeoutId)
  }, [copiedNumber])

  const fallbackCopyNumber = async (bank: string) => {
    const textArea = document.createElement('textarea')
    textArea.value = bank

    textArea.style.top = '0'
    textArea.style.left = '0'
    textArea.style.position = 'fixed'

    document.body.appendChild(textArea)
    textArea.focus()
    textArea.select()

    try {
      await document.execCommand('copy')
      textArea.style.display = 'none'
      setCopiedNumber(true)
    } catch (e) {
      console.error(e)
      setError(true)
    }
  }

  const copyNumber = async (bank: string) => {
    setBank(bank)

    if (!navigator.clipboard) {
      fallbackCopyNumber(bank === 'cimb' ? '800077521000' : '0600311611')
      return
    }

    try {
      await navigator.clipboard.writeText(
        bank === 'cimb' ? '800077521000' : '0600311611'
      )
      setCopiedNumber(true)
    } catch (e) {
      console.error(e)
      setError(true)
    }
  }

  const onErrorClose = (_: SyntheticEvent<Element, Event>, reason?: string) => {
    if (reason === 'clickaway') return
    setError(false)
  }

  return (
    <>
      <Head>
        <title>Persembahan | FreedomLife</title>
      </Head>

      <div className="container" style={{ paddingBottom: 30 }}>
        <h2 className="header__title--persembahan" data-testid="headerTitle">
          Persembahan &amp; Perpuluhan
        </h2>
        <p className="header__subtitle" data-testid="headerSubtitle">
          Gereja Kristen Kemah Daud - Yogyakarta
        </p>
      </div>

      <div className="container--fluid">
        <Card className="guidecard">
          <CardContent className="persembahan" style={{ padding: '16px 10%' }}>
            <p className="persembahan__text" data-testid="persembahanBrief">
              Dukungan untuk pekerjaan pelayanan Gereja Kristen Kemah Daud
              Yogyakarta dapat dilakukan melalui scan QRIS dibawah ini :
            </p>

            <img src="/images/qris.jpeg" alt="qris" style={{ width: '100%' }} />
            <p
              className="persembahan__text"
              style={{ fontSize: 13 }}
              data-testid="persembahanInfo"
            >
              Melalui aplikasi m-Banking / Digital Payment seperti :
            </p>
            <div
              className="persembahan__block"
              data-testid="persembahanInfo_Wallet"
            >
              <b>
                <i>Gopay, OVO, DANA &amp; LinkAja</i>
              </b>
            </div>
            <p
              className="persembahan__text"
              style={{ fontSize: 13, textAlign: 'center' }}
              data-testid="persembahanInfo_Admin"
            >
              (Bebas biaya administrasi untuk semua platform pembayaran)
            </p>

            <hr />
            <p className="persembahan__text" data-testid="persembahanInfo_Bank">
              Atau dapat juga dilakukan melalui transfer pada rekening bank
              dibawah ini :
            </p>

            <h1
              className="persembahan__text--bank"
              data-testid="persembahanInfo_CIMB"
            >
              CIMB Niaga
            </h1>
            <Button
              variant="outlined"
              className="persembahan__button"
              onClick={() => copyNumber('cimb')}
              startIcon={
                copiedNumber && bank === 'cimb' ? <SuccessIcon /> : null
              }
              data-testid="persembahanInfo_CIMB_btn"
            >
              {copiedNumber && bank === 'cimb'
                ? 'Nomor Tersalin!'
                : '8000-7752-1000'}
            </Button>
            <p
              className="persembahan__text"
              data-testid="persembahanInfo_CIMB_name"
            >
              a/n Gereja Kristen Kemah Daud (Klik Untuk Salin)
            </p>
            <h1
              className="bold-text"
              style={{ margin: '10px 0' }}
              data-testid="persembahanInfo_BCA"
            >
              BCA
            </h1>
            <Button
              variant="outlined"
              className="persembahan__button"
              onClick={() => copyNumber('bca')}
              startIcon={
                copiedNumber && bank === 'bca' ? <SuccessIcon /> : null
              }
              data-testid="persembahanInfo_BCA_btn"
            >
              {copiedNumber && bank === 'bca'
                ? 'Nomor Tersalin!'
                : '0600-311-611'}
            </Button>
            <p
              className="persembahan__text"
              data-testid="persembahanInfo_BCA_name"
            >
              a/n Yay Pelita Bangsa (Klik Untuk Salin)
            </p>
          </CardContent>
        </Card>
      </div>

      {/* Error Snackbar */}
      <Snackbar
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'center',
        }}
        open={error}
        autoHideDuration={3000}
        onClose={onErrorClose}
      >
        <Alert onClose={onErrorClose} severity="error">
          Terjadi kesalahan! Coba beberapa saat lagi.
        </Alert>
      </Snackbar>
    </>
  )
}

export default Persembahan
