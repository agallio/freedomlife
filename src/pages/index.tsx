import { useEffect, useState } from 'react'
import Router from 'next/router'
import { AnimatePresence } from 'framer-motion'

import JumboHeader from '@/components/JumboHeader'
import HomeBox from '@/components/Home/HomeBox'
import NewYearDialog from '@/components/Home/NewYearDialog'
import NewUserBox from '@/components/Home/NewUserBox'

import { useDispatchGuide } from '../store'

import dayjs from '@/utils/dayjs'
import useRequest from '@/utils/hooks/useRequest'

import type { ApiResponse, GuideDataResponse } from '@/types/api'

const Home = (): JSX.Element => {
  const guideDispatch = useDispatchGuide()
  const [open, setOpen] = useState(false)

  const { data } = useRequest<ApiResponse<GuideDataResponse>>({
    url: `/api/guide/${dayjs().format('DD-MM-YYYY')}`,
  })

  const toBible = () => {
    guideDispatch({ type: 'SET_GUIDE_DATE', data: '' })
    guideDispatch({ type: 'SET_GUIDE_PASSAGE', data: '' })
    Router.push('/read?guide=true')
    document.body.scrollTop = 0
    document.documentElement.scrollTop = 0
  }

  useEffect(() => {
    if (open) {
      document.body.style.overflow = 'hidden'
    } else {
      document.body.style.overflow = 'unset'
    }
  }, [open])

  return (
    <div className="max-w-md p-6 mx-auto mb-20 sm:py-6 md:mb-16 landscape:mx-auto">
      <JumboHeader isHome />

      <main>
        <HomeBox data={data} toBible={toBible} />

        <NewUserBox />

        <AnimatePresence>
          {open && <NewYearDialog handleClose={() => setOpen(false)} />}
        </AnimatePresence>
      </main>
    </div>
  )
}

export default Home
