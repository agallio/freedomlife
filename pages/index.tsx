import Router from 'next/router'
import {
  Card,
  CardContent,
  Fab,
  Fade,
  Grid,
  LinearProgress,
  Typography,
} from '@material-ui/core'
import { Skeleton } from '@material-ui/lab'

import { dayjs, useRequest } from '../src/utils'
import { useDispatchGuide } from '../src/store'

// Types
import type { ApiResponse, GuideDataResponse } from '../src/types'

export const Home: React.FC = () => {
  const guideDispatch = useDispatchGuide()

  const { data } = useRequest<ApiResponse<GuideDataResponse>>({
    url: '/api/guide/today',
  })

  const toBible = () => {
    guideDispatch({ type: 'SET_GUIDE_DATE', data: '' })
    Router.push('/bible')
  }

  return (
    <>
      {!data && <LinearProgress color="secondary" />}
      <Fade in>
        <div className="container" style={{ paddingBottom: 90 }}>
          <h2
            className="header__title"
            data-testid="headerTitle"
            style={{ marginTop: 65 }}
          >
            FreedomLife
          </h2>
          <p className="header__subtitle" data-testid="headerSubtitle">
            Aplikasi Panduan Baca Alkitab Setahun
          </p>

          <div style={{ marginTop: 40 }}>
            <Card className="homecard">
              <CardContent>
                <Typography
                  className="homecard__title"
                  data-testid="homecardTitle"
                  style={{ fontSize: '1.5rem !important', lineHeight: 1 }}
                >
                  Panduan Hari Ini
                </Typography>
                <Typography
                  className="homecard__subtitle"
                  data-testid="homecardSubtitle"
                  variant="subtitle1"
                >
                  {dayjs().format('dddd, DD MMMM YYYY')}
                </Typography>

                {['PL', 'PB', 'ALT'].map((item) => (
                  <Grid
                    key={item}
                    container
                    direction="row"
                    alignItems="center"
                    spacing={2}
                    style={{ marginTop: 5 }}
                  >
                    <Grid item xs={3} sm={2} md={2}>
                      <div className="homecard__pbox">
                        <p
                          className="homecard__pbox__text"
                          data-testid={`homecardPBox_${item}`}
                        >
                          {item}
                        </p>
                      </div>
                    </Grid>
                    <Grid item xs={9} sm={10} md={10}>
                      {!data ? (
                        <Skeleton variant="text" animation="wave" height={25} />
                      ) : (
                        <Typography
                          className="homecard__title"
                          variant="h6"
                          style={{ fontSize: 17 }}
                        >
                          {item === 'PL'
                            ? data.data?.pl_name
                            : item === 'PB'
                            ? data.data?.pb_name
                            : item === 'ALT'
                            ? data.data?.alt_name
                            : 'Tidak ada data'}
                        </Typography>
                      )}
                      <Typography
                        className="homecard__subtitle"
                        variant="subtitle1"
                        data-testid={`homecardPSub_${item}`}
                      >
                        {item === 'PL'
                          ? 'Perjanjian Lama'
                          : item === 'PB'
                          ? 'Perjanjian Baru'
                          : item === 'ALT'
                          ? 'Tambahan'
                          : ''}
                      </Typography>
                    </Grid>
                  </Grid>
                ))}

                <br />
                <Fab
                  className="homecard__button"
                  size="small"
                  variant="extended"
                  color="primary"
                  onClick={toBible}
                  disabled={!data}
                  data-testid="homecardButton"
                >
                  Baca
                </Fab>
              </CardContent>
            </Card>
          </div>
        </div>
      </Fade>
    </>
  )
}

export default Home
