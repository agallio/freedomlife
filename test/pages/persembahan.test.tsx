import { render } from '../testUtils'
import { Persembahan } from '../../pages/persembahan'

describe('Persembahan Page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Persembahan />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  describe('Header', () => {
    it('title rendered as "Persembahan & Perpuluhan"', () => {
      const { getByTestId } = render(<Persembahan />, {})
      const headerTitle = getByTestId('headerTitle')
      expect(headerTitle.textContent).toBe('Persembahan & Perpuluhan')
    })

    it('subtitle rendered as "Gereja Kristen Kemah Daud - Yogyakarta"', () => {
      const { getByTestId } = render(<Persembahan />, {})
      const headerSubtitle = getByTestId('headerSubtitle')
      expect(headerSubtitle.textContent).toBe(
        'Gereja Kristen Kemah Daud - Yogyakarta'
      )
    })
  })

  describe('Card', () => {
    it('brief text value is correct', () => {
      const { getByTestId } = render(<Persembahan />, {})
      const persembahanBrief = getByTestId('persembahanBrief')
      expect(persembahanBrief.textContent).toBe(
        'Dukungan untuk pekerjaan pelayanan Gereja Kristen Kemah Daud Yogyakarta dapat dilakukan melalui scan QRIS dibawah ini :'
      )
    })

    describe('Info', () => {
      it('info text value is correct', () => {
        const { getByTestId } = render(<Persembahan />, {})
        const persembahanInfo = getByTestId('persembahanInfo')
        expect(persembahanInfo.textContent).toBe(
          'Melalui aplikasi m-Banking / Digital Payment seperti :'
        )
      })

      it('e-wallet text value is correct', () => {
        const { getByTestId } = render(<Persembahan />, {})
        const persembahanInfo_Wallet = getByTestId('persembahanInfo_Wallet')
        expect(persembahanInfo_Wallet.textContent).toBe(
          'Gopay, OVO, DANA & LinkAja'
        )
      })

      it('admin fee text value is correct', () => {
        const { getByTestId } = render(<Persembahan />, {})
        const persembahanInfo_Admin = getByTestId('persembahanInfo_Admin')
        expect(persembahanInfo_Admin.textContent).toBe(
          '(Bebas biaya administrasi untuk semua platform pembayaran)'
        )
      })

      it('bank text value is correct', () => {
        const { getByTestId } = render(<Persembahan />, {})
        const persembahanInfo_Bank = getByTestId('persembahanInfo_Bank')
        expect(persembahanInfo_Bank.textContent).toBe(
          'Atau dapat juga dilakukan melalui transfer pada rekening bank dibawah ini :'
        )
      })

      it("CIMB bank rendered as 'CIMB Niaga'", () => {
        const { getByTestId } = render(<Persembahan />, {})
        const persembahanInfo_CIMB = getByTestId('persembahanInfo_CIMB')
        expect(persembahanInfo_CIMB.textContent).toBe('CIMB Niaga')
      })

      it('CIMB bank button value is correct', () => {
        const { getByTestId } = render(<Persembahan />, {})
        const persembahanInfo_CIMB_btn = getByTestId('persembahanInfo_CIMB_btn')
        expect(persembahanInfo_CIMB_btn.textContent).toBe('8000-7752-1000')
      })

      it('CIMB bank name is correct', () => {
        const { getByTestId } = render(<Persembahan />, {})
        const persembahanInfo_CIMB_name = getByTestId(
          'persembahanInfo_CIMB_name'
        )
        expect(persembahanInfo_CIMB_name.textContent).toBe(
          'a/n Gereja Kristen Kemah Daud (Klik Untuk Salin)'
        )
      })

      it("BCA bank rendered as 'BCA'", () => {
        const { getByTestId } = render(<Persembahan />, {})
        const persembahanInfo_BCA = getByTestId('persembahanInfo_BCA')
        expect(persembahanInfo_BCA.textContent).toBe('BCA')
      })

      it('BCA bank button value is correct', () => {
        const { getByTestId } = render(<Persembahan />, {})
        const persembahanInfo_BCA_btn = getByTestId('persembahanInfo_BCA_btn')
        expect(persembahanInfo_BCA_btn.textContent).toBe('0600-311-611')
      })

      it('BCA bank name is correct', () => {
        const { getByTestId } = render(<Persembahan />, {})
        const persembahanInfo_BCA_name = getByTestId('persembahanInfo_BCA_name')
        expect(persembahanInfo_BCA_name.textContent).toBe(
          'a/n Yay Pelita Bangsa (Klik Untuk Salin)'
        )
      })
    })
  })
})
