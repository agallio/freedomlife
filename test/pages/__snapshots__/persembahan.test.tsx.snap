// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Persembahan Page matches snapshot 1`] = `
<DocumentFragment>
  <div
    __source="[object Object]"
    class="container"
    style="padding-bottom: 30px;"
  >
    <h2
      __source="[object Object]"
      class="header__title--persembahan"
      data-testid="headerTitle"
    >
      Persembahan & Perpuluhan
    </h2>
    <p
      __source="[object Object]"
      class="header__subtitle"
      data-testid="headerSubtitle"
    >
      Gereja Kristen Kemah Daud - Yogyakarta
    </p>
  </div>
  <div
    __source="[object Object]"
    class="container--fluid"
  >
    <div
      __source="[object Object]"
      class="MuiPaper-root MuiCard-root guidecard MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        __source="[object Object]"
        class="MuiCardContent-root persembahan"
        style="padding: 16px 10%;"
      >
        <p
          __source="[object Object]"
          class="persembahan__text"
          data-testid="persembahanBrief"
        >
          Dukungan untuk pekerjaan pelayanan Gereja Kristen Kemah Daud Yogyakarta dapat dilakukan melalui scan QRIS dibawah ini :
        </p>
        <img
          __source="[object Object]"
          alt="qris"
          src="/images/qris.jpeg"
          style="width: 100%;"
        />
        <p
          __source="[object Object]"
          class="persembahan__text"
          data-testid="persembahanInfo"
          style="font-size: 13px;"
        >
          Melalui aplikasi m-Banking / Digital Payment seperti :
        </p>
        <div
          __source="[object Object]"
          class="persembahan__block"
          data-testid="persembahanInfo_Wallet"
        >
          <b
            __source="[object Object]"
          >
            <i
              __source="[object Object]"
            >
              Gopay, OVO, DANA & LinkAja
            </i>
          </b>
        </div>
        <p
          __source="[object Object]"
          class="persembahan__text"
          data-testid="persembahanInfo_Admin"
          style="font-size: 13px; text-align: center;"
        >
          (Bebas biaya administrasi untuk semua platform pembayaran)
        </p>
        <hr
          __source="[object Object]"
        />
        <p
          __source="[object Object]"
          class="persembahan__text"
          data-testid="persembahanInfo_Bank"
        >
          Atau dapat juga dilakukan melalui transfer pada rekening bank dibawah ini :
        </p>
        <h1
          __source="[object Object]"
          class="persembahan__text--bank"
          data-testid="persembahanInfo_CIMB"
        >
          CIMB Niaga
        </h1>
        <button
          __source="[object Object]"
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined persembahan__button"
          data-testid="persembahanInfo_CIMB_btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            8000-7752-1000
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <p
          __source="[object Object]"
          class="persembahan__text"
          data-testid="persembahanInfo_CIMB_name"
        >
          a/n Gereja Kristen Kemah Daud (Klik Untuk Salin)
        </p>
        <h1
          __source="[object Object]"
          class="bold-text"
          data-testid="persembahanInfo_BCA"
          style="margin: 10px 0px;"
        >
          BCA
        </h1>
        <button
          __source="[object Object]"
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined persembahan__button"
          data-testid="persembahanInfo_BCA_btn"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            0600-311-611
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <p
          __source="[object Object]"
          class="persembahan__text"
          data-testid="persembahanInfo_BCA_name"
        >
          a/n Yay Pelita Bangsa (Klik Untuk Salin)
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
